webhook_server/tests/test_github_api.py

```
@@ -1640,3 +1650,104 @@ def rmtree_fail(*args, **kwargs):
+                                    "'conventional-title' conflicts with built-in check" in call
+                                    for call in warning_calls
+                                )
+
+    def test_validate_custom_check_runs_duplicate_names(self, minimal_hook_data: dict, minimal_headers: dict) -> None:
+        """Test that duplicate custom check names are rejected."""
+        with patch("webhook_server.libs.github_api.Config") as mock_config:
+            mock_config.return_value.repository = True
+            mock_config.return_value.repository_local_data.return_value = {}
+
+            # Mock get_value to return custom checks with duplicate names
+            def get_value_side_effect(value: str, *_args: object, **_kwargs: object) -> Any:
```

Ruff (0.14.10)

Dynamically typed expressions (typing.Any) are disallowed in `get_value_side_effect`

(ANN401)